include/backup_restore.hpp

```diff
@@ -0,0 +1,96 @@
+#pragma once
+
+#include "types.hpp"
+
+#include <nlohmann/json.hpp>
+
+#include <tuple>
+
+namespace vpd
+{
+/**
+ * @brief class to implement backup and restore VPD.
+ *
+ */
+class BackupRestore
+{
+  public:
+    // delete functions
+    BackupRestore() = delete;
+    BackupRestore(const BackupRestore&) = delete;
+    BackupRestore& operator=(const BackupRestore&) = delete;
+    BackupRestore(BackupRestore&&) = delete;
+    BackupRestore& operator=(BackupRestore&&) = delete;
+
+    /**
+     * @brief Constructor.
+     *
+     * @param[in] l_parsedSystemJson - Parsed system config JSON.
+     *
+     */
+    BackupRestore(const nlohmann::json& l_parsedSystemJson);
+
+    /**
+     * @brief Defaul destructor.
+     */
+    ~BackupRestore() = default;
+
+    /**
+     * @brief An API to get backup and restore path information.
+     *
+     * This API is to get backup and restore's EEPROM or
+     * inventory path information present in the backup and restore config JSON.
+     *
+     * @param[in] i_tag - source or destination tag.
+     * @param[out] o_location - Holds hardware or inventory string value, for
+     * given i_tag.
+     * @param[out] o_path - Holds EEPROM or inventory path, for given
+     * i_tag.
+     */
+    void getBackupRestorePathInfo(const std::string& i_tag,
+                                  std::string& o_location, std::string& o_path);
+
+    /**
+     * @brief An API to backup and restore VPD.
+     *
+     * Note: This API works on the keywords declared in the backup and restore
+     * config JSON. Restore or backup action could be triggered for each
+     * keyword, based on the keyvalue present in the source and destination
+     * keyword.
+     *
+     * Restore source keyword value with destination keyword value,
+     * when source keyword has default value but
+     * destination has non default value.
+     *
+     * Backup the source keyword value on destination,
+     * when source keyword has non default value but
+     * destination has default value.
+     *
+     * @return Tuple of updated source and destination VPD map variant.
+     */
+    std::tuple<types::VPDMapVariant, types::VPDMapVariant> backupAndRestore();
+
+  private:
+    /**
+     * @brief An API to handle backup and restore IPZ VPD.
+     *
+     * @param[in,out] io_sourceVpdMap - Source VPD map.
+     * @param[in,out] io_dstVpdMap - Destination VPD map.
+     * @param[in] i_sourcePath - Source EEPROM file path or inventory path.
+     * @param[in] i_dstPath - Destination EEPROM file path or inventory path.
+     */
+    void backupAndRestoreIpzVpd(types::IPZVpdMap& io_sourceVpdMap,
+                                types::IPZVpdMap& io_dstVpdMap,
+                                const std::string& i_sourcePath,
+                                const std::string& i_dstPath);
+
+    // Parsed system config JSON
+    const nlohmann::json& m_parsedSystemJson;
```

Comment:
Why we need three class level variables?
only parsed back up and restore JSON object should be sufficient for now?

Reply:
removed as member variable. if required in later, we can bring it back.